napari/components/viewer_model.py

```diff
@@ -446,9 +446,9 @@ def add_volume(self, volume, *args, **kwargs):
         layer = layers.Volume(volume, *args, **kwargs)
         if self.dims.ndim == 2:
             self.dims.ndim = 3
-        self.dims.set_display(-3, True)
+        self.dims.set_display(self.dims.ndim - 3, True)
```

Comment:
shouldn't this be changed inside `self.dims.set_display` to perform this "wrap" when provided a negative number instead?

Reply:
fixed